### PR TITLE
Readme update

### PR DESCRIPTION
**Pull request by cube2222**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,7 +104,87 @@ Example output:
...
+
+Watermarks are a heuristic which try to approximate the "current time" when processing events. Said differently: When I receive a watermark for 12:00 I can be sure enough I've received all events of interest up to 12:00.
+
+To achieve this, they are generated at streaming sources and propagate downstream through the whole processing pipeline.
+
+The generation of watermarks usually relies on heuristics which provide satisfactory results for our given use case. OctoSQL currently contains the following watermark generators:
+- Maximum difference watermark generator (with an *offset* argument)
+  
+  With an offset of 10 seconds, this generator says: When I've received an event for 12:00:00, then I'm sure I won't receive any event older than 11:59:50.
+- Percentile watermark generator (with a *percentile* argument)
```
Comment (woojtek) [inline README.md:134]: actually 2 arguments, `precentile` and `frequency` but I guess you didn't mention it on purpose (?)

File: README.md [hunk 2]
```diff
@@ -104,7 +104,87 @@ Example output:
...
+  
+  With a percentile of 99.5, it will look at a specified number of recent events, and generate a watermark so that 99.5% of those events are before the watermark, and the remaining 0.5% are after it.
```
Comment (woojtek) [inline README.md:136]: no no no, if percentile is 99.5, then 99.5 of those events are AFTER the watermark, and the remaining 0.5 are BEFORE it

File: README.md [hunk 3]
```diff
@@ -104,7 +104,87 @@ Example output:
...
+This is where watermarks come into play.
+
+### Watermarks
+
+Watermarks are a heuristic which try to approximate the "current time" when processing events. Said differently: When I receive a watermark for 12:00 I can be sure enough I've received all events of interest up to 12:00.
+
+To achieve this, they are generated at streaming sources and propagate downstream through the whole processing pipeline.
+
+The generation of watermarks usually relies on heuristics which provide satisfactory results for our given use case. OctoSQL currently contains the following watermark generators:
+- Maximum difference watermark generator (with an *offset* argument)
```
Comment (woojtek) [inline README.md:131]: imho `argument` is much much better than *argument* and highlighted better, maybe switch it up that way (and compare the results)

File: README.md (hunk 1 above)
Comment (woojtek) [reply README.md:134]: same as above

File: README.md [hunk 4]
```diff
@@ -104,7 +104,87 @@ Example output:
...
+
+### Triggering
+Another matter is triggering of keys in aggregations.
+Sometimes you'd like to only see the value for a given key (hour) when you know it's done, but othertimes you'd like to see partial results (how's the unique user count going this hour).
+
+That's where you can use triggers. Triggers allow you to specify when a given aggregate (or join window for that matter) is emitted or updated. OctoSQL contains multiple triggers:
+- Watermark Trigger
+
+  This is the most straightforward trigger. It emits a value whenever the watermark for a given key (or the end of the stream) is reached. So basically the "show me when it's done".
+- Counting Trigger (with a *count* argument)
```
Comment (woojtek) [inline README.md:148]: same as above

File: README.md [hunk 5]
```diff
@@ -104,7 +104,87 @@ Example output:
...
+
+  This trigger will emit a value for a key every time it receives *count* records with this key. The count is reset whenever the key is triggered.
```
Comment (woojtek) [inline README.md:150]: same

File: README.md [hunk 6]
```diff
@@ -104,7 +104,87 @@ Example output:
...
+- Delay Trigger (with a *delay* argument)
```
Comment (woojtek) [inline README.md:151]: same

File: README.md [hunk 7]
```diff
@@ -104,7 +104,87 @@ Example output:
...
+
+  This trigger will emit a value for a key whenever the key has been inactive for the *delay* period.
```
Comment (woojtek) [inline README.md:153]: same

File: README.md [hunk 8]
```diff
@@ -104,7 +104,87 @@ Example output:
...
+
+You can use multiple triggers simultaneously. (Show me the current sum every 10 received events, but also the final value after having received the watermark.)
+
+### Retractions
+A key can be triggered multiple times with partial results. How do we know a given record is a retriggering of some key, and not a new unrelated record?
+
+OctoSQL solves this problem using a dataflow-like architecture. This means whenever a new value is sent for a key, a retraction is send for the old value. In practice this means every update is accompanied by the old record with an *undo* flag set.
```
Comment (woojtek) [inline README.md:160]: same

File: README.md [hunk 9]
```diff
@@ -104,7 +104,87 @@ Example output:
...
+### Example
+Here we can see how it all fits together:
+```sql
+WITH
+      with_watermark AS (SELECT *
+                         FROM max_diff_watermark(source=>TABLE(events),
+                                                 offset=>INTERVAL 5 SECONDS,
+                                                 time_field=>DESCRIPTOR(time)) e),
+      with_tumble AS (SELECT *
+                      FROM tumble(source=>TABLE(with_watermark),
```
Comment (woojtek) [inline README.md:173]: you are using tumble feature without explaining it earlier, maybe add a quick instruction in the beginning of this section to explain it

File: README.md (hunk 1 above)
Comment (cube2222) [reply README.md:134]: Yep, I omit technicalities here. Those are better served by the actual documentation you wrote.

File: README.md (hunk 2 above)
Comment (cube2222) [reply README.md:136]: Oh yeah, thanks.

Comment (cube2222) [reply README.md:136]: No wait.

Comment (cube2222) [reply README.md:136]: What?

Comment (cube2222) [reply README.md:136]: I'll have to write this down on a piece of paper and analyze this ;P

File: README.md (hunk 9 above)
Comment (cube2222) [reply README.md:173]: I'll describe it below

File: README.md (hunk 2 above)
Comment (cube2222) [reply README.md:136]: Yeah, alright, I get it.

**Pull request merged**